File: Python Fundamentals/9. Regular Expression/extract_emails.py
import re

pattern = r"(^|(?<=\s))(?P<user>([A-Za-z\d]+[\.\-_]?)+)(?P<sep>@)(?P<host>([A-Za-z]+\-?)+\.([A-Za-z]+\-?)+(\.([A-Za-z]+\-?)+)*)"
text = input()

emails = re.finditer(pattern, text)

for email in emails:

    print(email.group())

File: JavaScript Fundamentals/JavaScript Fundamentals Regex/extractEmails.js
function solve(input) {
    let pattern = /(?<user>^|(?<=\s)[A-Za-z\d]+([\.\-_]*[A-Za-z\d]+)*)@(?<host>[A-Za-z]+(\-[A-Za-z]+)*(\.[a-z]+)+)(?:\b|(?=\s))/g;

    let match = pattern.exec(input);

    while (match !== null) {
        console.log(match[0]);
        match = pattern.exec(input);
    }

}

solve('Just send email to s.miller@mit.edu and j.hopking@york.ac.uk for more information.');